presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java

```diff
@@ -148,6 +149,11 @@ public boolean transitionToRunning()
         return stageState.setIf(RUNNING, currentState -> currentState != RUNNING && !currentState.isDone());
     }
 
+    public boolean transitionToFlushing()
```

Comment:
`transitionToRunning()` needs to be updated to not go backwards from flushing to running